### ISSUE TITLE
logs are displayed incorrectly

### ISSUE THREAD

**Issue by imddl** [alayouni/ansi-highlighter#2]
When logging with logback in below pattern:
_%clr(%d{yyyy-MM-dd HH:mm:ss.SSS}){faint} %clr([%1.-1p]) %clr(%t){magenta} %clr([%C{64}:%M\\(%L\\)]){cyan} %m%n${LOG_EXCEPTION_CONVERSION_WORD:-%wEx}_
I got some logs as below:
_[2m2017-07-12 10:02:17.088[0;39m [32m[I][0;39m [35mmain[0;39m [36m[o.s.boot.context.embedded.tomcat.TomcatEmbeddedServletContainer:start(201)][0;39m Tomcat started on port(s): 3701 (http)
[2m2017-07-12 10:02:17.093[0;39m [32m[I][0;39m [35mmain[0;39m [36m[org.springframework.boot.StartupInfoLogger:logStarted(57)][0;39m Started BillingServer in 6.148 seconds (JVM running for 7.494)_
It can be displayed pretty in IntelliJ Console. However it displayed not that good in the plugin.
![image](https://user-images.githubusercontent.com/10821778/28098637-f87218f2-66e9-11e7-9cdc-a6971653d8aa.png)

In summary, the issues are:
1. Some style related texts were not removed such as the "[0;39m" and etc.
2. Although color "faint" is not supported it should not be displayed as cyan.
3. The last part of logging %m%n is displayed as cyan. However that part was not defined by any color.

**Comment by alayouni** [comment]
Hi @imddl,
Thanks for reporting.
In fact since 39 is not a supported code the parser is ignoring the entire sequence, which was originally intended as a form of warning that such codes are not supported.
Referring to your 3rd point, since the entire sequence "[0;39m" was ignored the reset code (0) was not taken into consideration, which led to the continuation of the previous sequence.
With that said I'll try to alter the current implementation and mimic the console behavior by end of week hopefully.

**Comment by alayouni** [comment]
Issue fixed with release [1.2.2](https://plugins.jetbrains.com/plugin/9707-ansi-highlighter/update/36904), currently awaiting approval from JetBrains.

commit: [b4232b0](https://github.com/alayouni/ansi-highlighter/commit/b4232b053cee41bc82fd11180e2d2006d2f970f4)

If in a rush you can download and install the update manually from [here](https://github.com/alayouni/ansi-highlighter/blob/master/ansi-highlighter.jar).

Closing the issue for now.
